### PR TITLE
Ignore background app starts

### PR DESCRIPTION
**Pull request by laurit**

### CODE REVIEW

File: splunk-otel-android/src/main/java/com/splunk/rum/AppStartupTimer.java [hunk 1]
```diff
@@ -44,15 +57,76 @@ Span start(Tracer tracer) {
         return appStart;
     }
 
+    /** Called when Activity is created. */
+    void startUiInit() {
+        if (uiInitStarted || isStartedFromBackground) {
+            return;
+        }
+        uiInitStarted = true;
```
Comment (breedx-splk) [inline splunk-otel-android/src/main/java/com/splunk/rum/AppStartupTimer.java:65]: This isn't threadsafe...do we care? Can two activities get created at the same time and step on each other?

File: splunk-otel-android/src/main/java/com/splunk/rum/AppStartupTimer.java [hunk 2]
```diff
@@ -44,15 +57,76 @@ Span start(Tracer tracer) {
...
+        }
+    }
+
+    void setCompletionCallback(Runnable completionCallback) {
+        this.completionCallback = completionCallback;
+    }
+
     void end() {
-        if (overallAppStartSpan != null) {
+        if (overallAppStartSpan != null && !uiInitTooLate && !isStartedFromBackground) {
```
Comment (breedx-splk) [inline splunk-otel-android/src/main/java/com/splunk/rum/AppStartupTimer.java:78]: Are we guaranteed that `end()` is called from the same thread that might have assigned those fields? If not, should we make all those volatile as well?

**Review by breedx-splk — APPROVED** [review]
This looks good to me. Because this is fixing a lifecycle issue, I do want to make sure we're also extra mindful of any threading concerns.

File: splunk-otel-android/src/main/java/com/splunk/rum/AppStartupTimer.java (hunk 1 above)
Comment (laurit) [reply splunk-otel-android/src/main/java/com/splunk/rum/AppStartupTimer.java:65]: This should be called from the UI thread, I wouldn't expect this to be called from multiple threads.

File: splunk-otel-android/src/main/java/com/splunk/rum/AppStartupTimer.java (hunk 2 above)
Comment (laurit) [reply splunk-otel-android/src/main/java/com/splunk/rum/AppStartupTimer.java:78]: This should also be called from the UI thread. I make `isStartedFromBackground` volatile, could make others also if you feel it is better that way.

Comment (mateuszrzeszutek) [reply splunk-otel-android/src/main/java/com/splunk/rum/AppStartupTimer.java:78]: `isStartedFromBackground` is only accessed from the UI thread, and does not need to be volatile, right?
I'd say that we either make all of these fields volatile, or make them all non-volatile and leave a comment saying sth like "these 3 booleans are only accessed from the UI thread"

**Review by mateuszrzeszutek — APPROVED** [review]

File: splunk-otel-android/src/main/java/com/splunk/rum/AppStartupTimer.java (hunk 2 above)
Comment (laurit) [reply splunk-otel-android/src/main/java/com/splunk/rum/AppStartupTimer.java:81]: Makes sense, reverted the previous change and added comment instead.

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**